ports/nrf/hal/hal_twi.c

```diff
@@ -50,16 +52,18 @@ void hal_twi_master_init(NRF_TWI_Type * p_instance, hal_twi_init_t const * p_twi
     p_instance->FREQUENCY = hal_twi_frequency_lookup[p_twi_init->freq];
     p_instance->ENABLE    = (TWI_ENABLE_ENABLE_Enabled << TWI_ENABLE_ENABLE_Pos);
 }
-#include <stdio.h>
-void hal_twi_master_tx(NRF_TWI_Type  * p_instance,
-                       uint8_t         addr,
-                       uint16_t        transfer_size,
-                       const uint8_t * tx_data,
-                       bool            stop) {
+
+hal_twi_error_t hal_twi_master_tx(NRF_TWI_Type  * p_instance,
+                                  uint8_t         addr,
+                                  uint16_t        transfer_size,
```

Comment:
unrelated to this PR, but this uint16_t should be a size_t (saves a few bytes of code size and seems more correct)

Reply:
as param in the function call this did not change anything, as the first 4 params are pushed on r0-r3 on cortex-m. However, i will see what effect it has to change it all over in the function.

Reply:
You are right, changing number_of_rxd_bytes/number_of_txd_bytes to size_t shrinks the text down 8 bytes. I'll apply the patch.

Reply:
Yeah I think the saving occurs when calculations are done with uint16_t values, e.g. adding 1 to a uint16_t requires some extra code to wrap around, which will be done automatically with a 32-bit int. It's a bit counter-intuitive that a longer int is more efficient, but that's what `uint_fast16_t` and the likes are for.